### PR TITLE
Move State to fixture package

### PR DESCRIPTION
**Pull request by lavalamp**
Builds on #45.

I think importing with the . is potentially a little sketchy, we'll see how I feel in the morning. LMK what you think.

### CODE REVIEW

File: internal/fixture/state.go [hunk 1]
```diff
@@ -34,6 +35,53 @@ type State struct {
...
 }
 
+// FixTabsOrDie counts the number of tab characters preceding the first line in
+// the given yaml object. It removes that many tabs from every line. It then
+// converts remaining tabs to spaces (two spaces per tab). It panics (it's a
+// test funtion) if it finds mixed tabs and spaces in front of a line, or if
+// some line has fewer tabs than the first line.
+//
+// The purpose of this is to make it easier to read tests.
+func FixTabsOrDie(in typed.YAMLObject) typed.YAMLObject {
```
[the remark this answers is not in the document]
Comment (lavalamp) [reply internal/fixture/state.go]: It's in an internal package so it's not really that public. I think it might be useful in other tests, too.

**Comment by apelisse** [comment]
I don't really mind the dot as much as I don't like the name of the package. I also suspect that you want to use the dot because you don't like the name either :-). Can we have use ... `mergetest`, `updatetest`? I like testing packages to end in test, like `httptest` and other golang test packages.

**Comment by lavalamp** [comment]
I think fixture is an OK name (esp since it's internal), I used the dot because I'm lazy and also it'd add verbosity no matter what the name is.

**Pull request merged**